File: unittest/scripts/py_dev_api_examples/working_with_collections/Basic_CRUD_Operations_on_Collections.py

# Connecting to MySQL Server and working with a Collection

import mysqlx

# Connect to server
mySession = mysqlx.getSession( {
'host': 'localhost', 'port': 33060,
'dbUser': 'mike', 'dbPassword': 's3cr3t!'} )

myDb = mySession.getSchema('test')

# Create a new collection 'my_collection'
myColl = myDb.createCollection('my_collection')

# Insert documents
myColl.add({'name': 'Sakila', 'age': 15}).execute()
myColl.add({'name': 'Susanne', 'age': 24}).execute()
myColl.add({'name': 'Mike', 'age': 39}).execute()

# Find a document
docs = myColl.find('name like :param1 AND age < :param2').
          limit(1).
          bind('param1','S%').
          bind('param2',20).
          execute()

# Print document
doc = docs.fetchOne()
print doc

# Drop the collection
session.dropCollection('test','my_collection')

File: unittest/scripts/js_dev_api_examples/working_with_collections/Basic_CRUD_Operations_on_Collections.js

// Connecting to MySQL Server and working with a Collection

var mysqlx = require('mysqlx');

// Connect to server
var mySession = mysqlx.getSession( {
host: 'localhost', port: 33060,
dbUser: 'mike', dbPassword: 's3cr3t!'} );

var myDb = mySession.getSchema('test');

// Create a new collection 'my_collection'
var myColl = myDb.createCollection('my_collection');

// Insert documents
myColl.add({_id: '1', name: 'Sakila', age: 15}).execute();
myColl.add({_id: '2', name: 'Susanne', age: 24}).execute();
myColl.add({_id: '3', name: 'Mike', age: 39}).execute();

// Find a document
var docs = myColl.find('name like :param1 AND age < :param2').limit(1).
        bind('param1','S%').bind('param2',20).execute();

// Print document
print(docs.fetchOne());

// Drop the collection
myDb.dropCollection('my_collection');
